cloudstack.go

```diff
@@ -410,10 +417,11 @@ func (d *Driver) Create() error {
 		return err
 	}
 	d.Id = vm.Id
-	d.PrivateIP = vm.Nic[0].Ipaddress
+	d.PrivateIP = vm.Nic[d.NetworkInterface].Ipaddress
```

Comment:
It would be nice to add some bounds checking here to ensure we don't raise a panic at runtime. I can think of two options on handling this, either we could simply allow `d.PrivateIP` to be empty or return an error if it's out of bounds.
No strong options but I think returning an error would allow us to catch errors more easily.